tests/unit/datarobot_drum/drum/adapters/model_adapters/test_python_model_adapter.py

```diff
@@ -323,3 +325,80 @@ def test_multiple_artifacts_detection_negative(self):
             Path(f"{dir_name}/file2.pkl").touch()
             with pytest.raises(DrumCommonException, match="Multiple serialized model files found."):
                 adapter._detect_model_artifact_file()
+
+
+class TestPredictResultSplitter:
+    """
+    Test the method that takes the predict output DataFrame and splits it to predictions DataFrame
+    and extra DataFrame.
+    """
+
+    @pytest.fixture
+    def num_rows(self):
+        return 3
+
+    @pytest.fixture
+    def binary_df(self, num_rows):
+        positive_probabilities = [float(random.random()) for _ in range(num_rows)]
+        negative_probabilities = [1 - v for v in positive_probabilities]
+        return pd.DataFrame({"0": negative_probabilities, "1": positive_probabilities})
+
+    @pytest.fixture
+    def multiclass_df(self, num_rows):
+        class1_values = [0.1 * index for index in range(1, num_rows + 1)]
+        class2_values = [0.2 * index for index in range(1, num_rows + 1)]
+        class3_values = [1 - (c1 + c2) for c1, c2 in zip(class1_values, class2_values)]
+        data = {"class1": class1_values, "class2": class2_values, "class3": class3_values}
+        return pd.DataFrame(data)
+
+    @pytest.fixture
+    def regression_df(self, num_rows):
+        return pd.DataFrame({"Predictions": [random.random() for _ in range(num_rows)]})
+
+    def test_split_result_of_binary_classification(self, binary_df):
+        pred_df, extra_df = PythonModelAdapter._split_to_predictions_and_extra_df(
+            binary_df, request_labels=binary_df.columns.tolist()
+        )
+        assert pred_df.equals(binary_df)
+        assert extra_df is None
+
+    def test_split_result_of_binary_classification_with_extra(self, num_rows, binary_df):
```

Comment:
cc @scottp-ml

Reply:
The changes in this PR should not apply to the new interface. It is only for custom models.
Personally, my opinion is that private methods can have unit-tests. Sometimes and attempt to only test public methods results in many mocks and patches, while having unit tests to private methods keep the tests lean and mean.

Reply:
that seems ... incorrect, i think.  the call stack is

- score request
- datarobot_drum.resource.components.Python.generic_predictor.generic_predictor.GenericPredictorComponent.configure  (this is where we set the predictor)
- datarobot_drum.resource.components.Python.generic_predictor.generic_predictor.GenericPredictorComponent._materialize
- datarobot_drum.drum.language_predictors.base_language_predictor.BaseLanguagePredictor.predict
- datarobot_drum.drum.language_predictors.python_predictor.python_predictor.PythonPredictor._predict
- datarobot_drum.drum.adapters.model_adapters.python_model_adapter.PythonModelAdapter.predict
- then either:  
- datarobot_drum.drum.adapters.model_adapters.python_model_adapter.PythonModelAdapter._predict_new_drum
- datarobot_drum.drum.adapters.model_adapters.python_model_adapter.PythonModelAdapter._predict_legacy_drum

the generic predictor now want a three-part tuple, which everything below it on the stack should be returning and they're not.  I'm pretty sure you missed `datarobot_drum.drum.language_predictors.python_predictor.python_predictor.PythonPredictor._predict` and `datarobot_drum.drum.adapters.model_adapters.python_model_adapter.PythonModelAdapter._predict_new_drum`


This would be much easier to follow by using a dataclass to return these values and then type-hinting all up and down the call stack.

Reply:
Done